### DISCUSSION TITLE
Block relative

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#1771] [Help] [answered]
Hey, when Im using this code: 
```
val blockPosition = event.blockPosition.relative(event.blockFace.oppositeFace);
        val block = event.instance.getBlock(blockPosition)
        val water = Block.WATER.withTag(Tag.Boolean("water"), true)
        println(event.block)
        println(block)
        println(event.player.itemInMainHand.material())
```
Oh, and event.block comes from the PlayerBlockInteractEvent event.
it will give me the following result in my console:
![image](https://user-images.githubusercontent.com/71212639/222984981-ef424872-17f3-40c3-9b5b-df9dc863384e.png)

But this shouldnt be correct. I've place the water on top of the yellow wool as you can see in the picture below.
![image](https://user-images.githubusercontent.com/71212639/222985012-0d05c4ff-f07b-4d40-8e04-6b0716160b77.png)

What is wrong here? Im getting the cobble stone instead of the water.

**Comment by GoldenCookieGIT** [comment, marked as answer]
if you clicked on top of the wool, the blockface would be `TOP` and since youre getting the block opposite of the block face itll get it from blockface `BOTTOM`

**Reply by DevBlokJochem** [reply]
Hmm, XD
if I want to get the water block. which function do I have to use instead of .relative?

**Reply by GoldenCookieGIT** [reply]
use event.blockFace, you don't need the opposite face

**Reply by DevBlokJochem** [reply]
oh, my mistake. I was thinking to hard about the problem that I didn't see that.
thank you
